### PR TITLE
HDDS-6148. Validate ContainerBalancerConfiguration before start ContainerBalancer

### PR DESCRIPTION
**Pull request by symious**
## What changes were proposed in this pull request?

In HDDS-6070, we did some refactoring on ContainerBalancerConfiguration, and we added a method called validateConfiguration to check the configurations. 

Currently the method returns "void", but it's more reasonable to return a boolean to indicate the validate result of the configurations.

This ticket is to change the return parameter from "void" to boolean to check the parameters before doing container balancer work.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6148

## How was this patch tested?

unit test

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java [hunk 1]
```diff
@@ -157,7 +157,9 @@ public boolean start(ContainerBalancerConfiguration balancerConfiguration) {
 
       balancerRunning = true;
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java:158]: I don't think this should be set if config validation fails.

Comment (symious) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java]: @adoroszlai Thanks for the review. Change to set the value after validation.

**Comment by symious** [comment]
@siddhantsangwan @ferhui @JacksonYao287 Could you help to review this PR?

**Comment by siddhantsangwan** [comment]
@symious Changes look good to me!

**Comment by symious** [comment]
@siddhantsangwan Thanks for the review.

**Comment by ferhui** [comment]
@symious Thanks for your contribution. @siddhantsangwan @adoroszlai Thanks for your reviews! Merged

**Comment by ferhui** [comment]
Forget to leave review msg. It looks good to me. +1

**Pull request merged**